src/mqt/yaqs/core/data_structures/simulation_program.py

```
@@ -0,0 +1,702 @@
+        TypeError: If ``segments`` or an item has the wrong type.
+        ValueError: If ``segments`` is empty, a segment sets program-owned fields,
+            or ``num_traj`` is less than one.
+    """
+
+    segments: tuple[_ProgramSegment, ...]
+    observables: tuple[Observable, ...]
+    num_traj: int | None
+    random_seed: int | None
+    get_state: bool
+
+    def __init__(
```

Ruff (0.16.1)

[warning] Too many branches (13 > 12)

(PLR0912)